### PR TITLE
Vikash branch 2

### PR DESCRIPTION
**Pull request by mkaii**

### CODE REVIEW

File: class-3-branches-reset/commands.txt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 git remote add <alias_name> <.git url>
 git remote rm <alias_name>
+git remote -v 
 git branch
 git checkout -b <branch_name>
 git checkout <branch_name>
-git switch <branch_name>
\ No newline at end of file
+git switch <branch_name>
+git reset <commit_id>  --hard/soft/mixed
```
Comment (mkaii) [inline class-3-branches-reset/commands.txt]: Hey @vikash123765 , the commands seem correct but a small suggestion that -- needs to be in all the modes and not just in hard
Fix :  git reset <commit_id> --hard/--soft/--mixed

also note that u dont necessarily need the commit id, U may also use the head pointer of a branch
Fix : git reset head~1 --hard/--soft/--mixed

File: class-3-branches-reset/commands.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 git remote rm <alias_name>
+git remote -v 
 git branch
 git checkout -b <branch_name>
 git checkout <branch_name>
-git switch <branch_name>
\ No newline at end of file
+git switch <branch_name>
+git reset <commit_id>  --hard/soft/mixed 
+git rebase <branch_you_want_to_copy>
```
Comment (mkaii) [inline class-3-branches-reset/commands.txt:10]: @vikash123765 
please make a note that this command brings the changes from the branch mentioned in the command
to the branch u are on right note 
PS: copy seems ambiguous to me as a reader.

File: class-3-branches-reset/commands.txt [hunk 3]
```diff
@@ -1,6 +1,9 @@
 git remote add <alias_name> <.git url>
 git remote rm <alias_name>
+git remote -v 
 git branch
 git checkout -b <branch_name>
```
Comment (mkaii) [inline class-3-branches-reset/commands.txt:5]: @vikash123765 Note : u could used git branch <branch_name> to create the branch but u have to shift manually using checkout or switch

File: class-3-branches-reset/commands.txt (hunk 1 above)
Comment (vikash123765) [inline class-3-branches-reset/commands.txt]: I dont have tilde.

File: class-4-merge-stashed-staged_diff/class_4_commands .txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+git merge <commit_id>  = like rebase but commit structure may change
+git cherry-pick <commit_id> = can apply only commit to branch insted merging or rebase 
+git stash -u = will save change in stack to be retrived when we want.will not show in staged or untracked 
+git stash apply = retrieve change from stack.
```
Comment (mkaii) [inline class-4-merge-stashed-staged_diff/class_4_commands .txt]: Note : this works in last in first out fashion

File: class-4-merge-stashed-staged_diff/class_4_commands .txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+git diff -staged = will show changes made in file.
```
Comment (mkaii) [inline class-4-merge-stashed-staged_diff/class_4_commands .txt]: Fix : git diff --staged

File: class-4-merge-stashed-staged_diff/class_4_commands .txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+git diff --staged <path> <file-name.patch> = will create a patch file with the changes made of we can send and somone can apply.
```
Comment (mkaii) [inline class-4-merge-stashed-staged_diff/class_4_commands .txt]: git diff --staged> path/<anyName.PATCH>

**Review by mkaii — COMMENTED** [review]
@vikash123765 reviewed, please incorporate the mentioned fixes.

**Pull request merged**